In [1]:
%matplotlib inline
import pandas as pd
import geopandas as gpd
import plotly.express as px
import plotly
import matplotlib.pyplot as plt
import pandas as pd
from IPython.display import display
from IPython.display import Image
import pandas as pd
import numpy as np

In [2]:
healthy_corner_stores_geo_df=gpd.read_file('../../data/Raw_Data/PhillyHealth_Healthy_corner_stores.geojson')

* This dataset shows the locations of healthy corner stores in Philly. By plotting these dots against other maps such as heatmaps of income and diabetes rates, I can see the relationship between access to healthy corner stores and these other variables. As I analyze my data in more depth and begin to produce my own data, I can merge this data with a dataset showing locations of farmers markets to get a more comprehensive dataset surrounding overall access to healthy foods.

# Exploration

In [3]:
healthy_corner_stores_geo_df.shape

(669, 6)

* There are 669 healthy corner stores documented in this dataset and 6 columns of details pertaining to each corner store. 

In [4]:
healthy_corner_stores_geo_df.head()

CATEGORY            STORE_ADDR               OFFICIAL_S    ZIP  CDC_STORE_  \
0     Basic         1739 South St          Adabel Deli Inc  19146         1.0   
1  Enhanced     1643 W Diamond St             Adan Grocery  19121         3.0   
2     Basic    2568 N Bancroft St  Agramonte Grocery Store  19132         2.0   
3  Enhanced        4600 N 11th St    Ailyn Mini Market LLC  19140         3.0   
4     Basic  419 East Wyoming Ave            Al Aqsa Halal  19120         2.0   

                                       geometry  
0  POINT (-75.17094707694254 39.94437406209261)  
1  POINT (-75.16048543262013 39.98528161550929)  
2  POINT (-75.15885414520487 39.99238311557681)  
3  POINT (-75.14294922174258 40.02234870746425)  
4  POINT (-75.11744231378752 40.02075034319167)

In [5]:
healthy_corner_stores_geo_df.columns

Index(['CATEGORY', 'STORE_ADDR', 'OFFICIAL_S', 'ZIP', 'CDC_STORE_',
       'geometry'],
      dtype='object')

* The information most relevant to me would be the primarily the __geometry, address, names, and zipcodes__. 

In [6]:
healthy_corner_stores_geo_df['geometry'].head()

0    POINT (-75.17094707694254 39.94437406209261)
1    POINT (-75.16048543262013 39.98528161550929)
2    POINT (-75.15885414520487 39.99238311557681)
3    POINT (-75.14294922174258 40.02234870746425)
4    POINT (-75.11744231378752 40.02075034319167)
Name: geometry, dtype: object

In [7]:
healthy_corner_stores_geo_df['STORE_ADDR'].head()

0           1739 South St
1       1643 W Diamond St
2      2568 N Bancroft St
3          4600 N 11th St
4    419 East Wyoming Ave
Name: STORE_ADDR, dtype: object

In [8]:
healthy_corner_stores_geo_df['OFFICIAL_S'].head()

0            Adabel Deli Inc
1               Adan Grocery
2    Agramonte Grocery Store
3      Ailyn Mini Market LLC
4              Al Aqsa Halal
Name: OFFICIAL_S, dtype: object

In [9]:
healthy_corner_stores_geo_df['ZIP'].unique()

array(['19146', '19121', '19132', '19140', '19120', '19145', '19142',
       '19134', '19133', '19130', '19139', '19124', '19143', '19131',
       '19137', '19135', '19125', '19147', '19144', '19122', '19148',
       '19111', '19141', '19151', '19104', '19138', '19123', '19149',
       '19152', '19136', '19128', '19115', '19126', '19129'], dtype=object)

In [10]:
healthy_corner_stores_geo_df['ZIP'].nunique()

34

* I can see here there are 34 zipcodes accounted for in this dataset.

In [11]:
healthy_corner_stores_geo_df['ZIP'].value_counts()

19134    68
19132    47
19140    46
19120    43
19143    43
19148    39
19133    38
19121    34
19146    27
19124    26
19139    26
19142    26
19145    22
19144    19
19131    18
19122    16
19147    16
19138    14
19151    13
19123    13
19130    11
19141     8
19125     8
19136     7
19104     7
19135     7
19111     7
19126     6
19128     5
19149     3
19152     2
19137     2
19115     1
19129     1
Name: ZIP, dtype: int64

These are the number of healthy corner stores in each of the zipcodes. Since I will be plotting these later (combined with farmers markets) it helps to get familiar with thinking about the number of locations within each zipcode. 

# Cleaning

In [12]:
cleaned_healthycornerstores_df=healthy_corner_stores_geo_df[['STORE_ADDR', 'OFFICIAL_S', 'ZIP', 'geometry']]
cleaned_healthycornerstores_df.head()

STORE_ADDR               OFFICIAL_S    ZIP  \
0         1739 South St          Adabel Deli Inc  19146   
1     1643 W Diamond St             Adan Grocery  19121   
2    2568 N Bancroft St  Agramonte Grocery Store  19132   
3        4600 N 11th St    Ailyn Mini Market LLC  19140   
4  419 East Wyoming Ave            Al Aqsa Halal  19120   

                                       geometry  
0  POINT (-75.17094707694254 39.94437406209261)  
1  POINT (-75.16048543262013 39.98528161550929)  
2  POINT (-75.15885414520487 39.99238311557681)  
3  POINT (-75.14294922174258 40.02234870746425)  
4  POINT (-75.11744231378752 40.02075034319167)

* This condensed version shows me only the columns I care about. 

In [13]:
cleaned_healthycornerstores_df.columns = ['Address', 'Name', 'Zipcode', 'geometry']

In [14]:
cleaned_healthycornerstores_df.head()

Address                     Name Zipcode  \
0         1739 South St          Adabel Deli Inc   19146   
1     1643 W Diamond St             Adan Grocery   19121   
2    2568 N Bancroft St  Agramonte Grocery Store   19132   
3        4600 N 11th St    Ailyn Mini Market LLC   19140   
4  419 East Wyoming Ave            Al Aqsa Halal   19120   

                                       geometry  
0  POINT (-75.17094707694254 39.94437406209261)  
1  POINT (-75.16048543262013 39.98528161550929)  
2  POINT (-75.15885414520487 39.99238311557681)  
3  POINT (-75.14294922174258 40.02234870746425)  
4  POINT (-75.11744231378752 40.02075034319167)

* This has titles that are nicer-looking and more accurate. 
* This cleaned_healthycornerstores_df will be the df I use for healthy store counts when I need it in the form of Coordinates as Points (for my dot maps), rather than in the form of latitude and longitude for interactive maps. 

### Splitting Coordinates into Latitude and Longitude for Interactive Maps

* I also plan to an interactive map of the farmers markets locations, so I will need to separate the longitude and latitude into different columns.

In [15]:
cleaned_healthycornerstores_df['longitude']=cleaned_healthycornerstores_df['geometry'].x

/usr/local/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning:


A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy



In [16]:
cleaned_healthycornerstores_df['latitude']=cleaned_healthycornerstores_df['geometry'].y

In [17]:
cleaned_healthycornerstores_df.head()

Address                     Name Zipcode  \
0         1739 South St          Adabel Deli Inc   19146   
1     1643 W Diamond St             Adan Grocery   19121   
2    2568 N Bancroft St  Agramonte Grocery Store   19132   
3        4600 N 11th St    Ailyn Mini Market LLC   19140   
4  419 East Wyoming Ave            Al Aqsa Halal   19120   

                                       geometry  longitude   latitude  
0  POINT (-75.17094707694254 39.94437406209261) -75.170947  39.944374  
1  POINT (-75.16048543262013 39.98528161550929) -75.160485  39.985282  
2  POINT (-75.15885414520487 39.99238311557681) -75.158854  39.992383  
3  POINT (-75.14294922174258 40.02234870746425) -75.142949  40.022349  
4  POINT (-75.11744231378752 40.02075034319167) -75.117442  40.020750